### PR TITLE
Handle multiple GraphQL queries run in a single HTTP request

### LINKED ISSUE #47: optics not fully compatible with transport-level batching
The optics agent instrumentation assumes only one query operation per HTTP request.

At the very least, this is implied where the schema-level resolve function takes its `info` and puts it in a single `info` slot on the HTTP-request-level `opticsContext` object.

There are two ways that this can be broken:
- Using apollo-client/server's transport-level batching, multiple "query documents" get sent on the same HTTP request
- You can always specify multiple "query operations" in one "query document". (Though, eg, react-apollo forbids this)

My guess is that the schema-level resolve function should add its info to a list on opticsContext. reportRequestEnd should iterate over them. reportRequestStart and reportTrace should be refactored to take the specific info they care about as a parameter rather than (or in addition to) the opticsContext.

This likely causes the issues we've seen where fields not mentioned in the query are sent to the optics backend as nodes.

### PR DESCRIPTION
**Pull request by n1mmy**
(aka transport level batching)

Note: due to indentation changes, this is likely easier to review with `git diff -w`

Fixes #47

### CODE REVIEW

**Comment by glasser** [comment]
comments from slack (boo DNS):

- use Map to avoid quadratic
- put resolvers:[] on query object from the start
- two typos

File: src/Report.js [hunk 1]
```diff
@@ -394,14 +394,25 @@ export const reportRequestStart = (context, queryInfo, queryContext) => {
...
-    context.queries = []; // eslint-disable-line no-param-reassign
+    context.queries = new Map(); // eslint-disable-line no-param-reassign
   }
-  context.queries.push({
+  if (!context.queries.has(queryContext)) {
+    context.queries.set(queryContext, []);
+  }
+  context.queries.get(queryContext).push({
     info: queryInfo,
     context: queryContext,
```
Comment (glasser) [inline src/Report.js]: Is this field used?

File: src/Report.js [hunk 2]
```diff
@@ -466,7 +473,9 @@ export const reportRequestEnd = (req) => {
 
     // Iterate over each query in this request and aggregate its
     // timing and resolvers.
-    queries.forEach(({ info, resolvers: queryResolvers = [] }) => {
+    const flatQueries = [];
+    queries.forEach(qList => qList.forEach(qObj => flatQueries.push(qObj)));
```
Comment (glasser) [inline src/Report.js]: Not really sure why this is better than just doing nested iteration...

Comment (n1mmy) [inline src/Report.js]: Avoids another level of indentation and makes it more clear we're just iterating over all the queries and not doing anything fancy. I can change it if you like.

**Pull request merged**